lib/Dialect/TTMetal/Transforms/Passes.cpp

```diff
@@ -175,12 +187,57 @@ class TTIRToTTMetalLayoutRewriter : public OpRewritePattern<ttir::ToLayoutOp> {
     assert(inputLayout.getPhysicalShape(inputTy.getShape()) ==
                outputLayout.getPhysicalShape(outputTy.getShape()) &&
            "Physical shapes must match for now");
-    AffineMap src =
-        inputLayout.projectOnto(inputTy.getShape(), device.getGrid());
-    AffineMap dst =
-        outputLayout.projectOnto(outputTy.getShape(), device.getGrid());
-    auto dm = calculateDataMovement(
-        inputTy.getShape(), inputLayout.getElementSizeBytes(), src, dst);
+
+    // Shape that will be traversed when calculating the data movement between
+    // layouts.
+    SmallVector<int64_t> inputShape =
+        inputLayout.isTiled() ? inputLayout.getTiledShape(inputTy.getShape())
+                              : SmallVector<int64_t>(inputTy.getShape());
+    SmallVector<int64_t> outputShape =
+        outputLayout.isTiled() ? outputLayout.getTiledShape(outputTy.getShape())
+                               : SmallVector<int64_t>(outputTy.getShape());
+
+    AffineMap inputLinearMap = inputLayout.isTiled()
+                                   ? inputLayout.getTileLinearMap()
+                                   : inputLayout.getLinear();
+    AffineMap outputLinearMap = outputLayout.isTiled()
+                                    ? outputLayout.getTileLinearMap()
+                                    : outputLayout.getLinear();
+
+    if (inputLayout.isTiled()) {
+      assert(inputLinearMap == outputLinearMap &&
+             "Linear maps must be identical");
```

Comment:
Ah ok I missed this too, I think the assert needs to be moved up before line 200.

```
assert(!(inputLayout.isTiled() && outputLayout.isTiled()) || inputLayout.getLinear() == outputLayout.getLinear());
```

In other words, if we're doing tile reblocking, the scalar affine maps must match.